selfupdate.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/gokrazy/updater"
+)
+
+type updateRequest struct {
+	MachineID string `json:"machine_id"`
+	SBOMHash  string `json:"sbom_hash"`
+}
+
+type updateResponse struct {
+	SBOMHash     string `json:"sbom_hash"`
+	RegistryType string `json:"registry_type"`
+	Link         string `json:"download_link"`
+}
+
+func checkForUpdates(ctx context.Context, request *updateRequest) (*updateResponse, error) {
+	reqBody, err := json.Marshal(request)
+	if err != nil {
+		return nil, fmt.Errorf("error json encoding request: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", updateEndpoint, bytes.NewBuffer(reqBody))
+	if err != nil {
+		return nil, fmt.Errorf("error creating http request: %w", err)
+	}
+	req.Header.Set("Content-Type", "text/json")
```

Comment:
We should set both headers: Accept and Content-Type. 

The MIME type is application/json, not text/json